### ISSUE TITLE
How to update uploaded files?

### ISSUE THREAD

**Issue by cdupont** [cdupont/Nomyx#101]
The current game allows to upload files containing helper functions and rules.
Once uploaded, those files cannot be updated: some rules inside the files could have been proposed/activated and changing the content of the file would change the behavior of the game without voting.
However it should be possible to update those files in the process of preparing rules.
So, question is: can we detect if a file content is used in a proposed rule? If yes, overwriting the file would be forbidden. Otherwise, it would be authorised.

**Comment by cdupont** [comment]
The current system asks the player to put a version number on their upload: SpaceshipV1.hs.
Then the usage should be qualified: SpaceshipV1.myRule.
This system is a bit clunky and could be automatized, with or without git support.
Without Git:

- files versions are appended automatically a version number when uploaded: module Spaceship becomes SpaceshipV1

With Git:

- the player can configure a GitHub repo within Nomyx (#90)
- Nomyx would download the history of all files
- all the versions with a tag Nomyx-XX will be renamed (module Spaceship becomes SpaceshipXX)
- those versions only are imported and put in scope

So basically, if you want to propose a rule from your github rule set, you need to tag it appropriately. Only then it will appear in Nomyx.

**Comment by cdupont** [comment]
Another proposition is to put the uploaded files inside the rules. Uploaded files are considered as an annexe of the rule and is voted on altogether. This solves the problem in #

**Comment by cdupont** [comment]
Done in Library
